Avoids recursion during panic!?

Similar to (but not the same):
https://bugzilla.mozilla.org/show_bug.cgi?id=1435219

diff -r 95fd9deac911 js/src/wasm/WasmSignalHandlers.cpp
--- a/js/src/wasm/WasmSignalHandlers.cpp	Mon Feb 05 12:14:22 2018 +0100
+++ b/js/src/wasm/WasmSignalHandlers.cpp	Mon Feb 05 05:22:59 2018 -0800
@@ -181,6 +181,11 @@
 #  define FP_sig(p) ((p)->uc_mcontext.mc_fp)
 #  define SP_sig(p) ((p)->uc_mcontext.mc_i7)
 # endif
+# if defined(__sun__) && defined(__sparc__)
+#  define PC_sig(p) ((p)->uc_mcontext.gregs[REG_PC])
+#  define FP_sig(p) ((p)->uc_mcontext.gregs[REG_FPRS])
+#  define SP_sig(p) ((p)->uc_mcontext.gregs[REG_SP])
+# endif
 #elif defined(__NetBSD__)
 # define XMM_sig(p,i) (((struct fxsave64*)(p)->uc_mcontext.__fpregs)->fx_xmm[i])
 # define EIP_sig(p) ((p)->uc_mcontext.__gregs[_REG_EIP])
